apps/docs/content/guides/functions/debugging.mdx

```diff
@@ -156,6 +160,19 @@ Examine if there are any unnecessary or redundant dependencies that can be remov
 deno info --import-map=/path/to/import_map.json /path/to/function/index.ts
 ```
 
+#### NPM Dependencies
+
+Additionally, if you utilize NPM modules in your Edge Functions, it's crucial to be mindful of their size and impact on the overall bundle size. While importing NPM modules, consider using the notation `import { submodule } from 'npm:package/submodule'` to selectively import specific submodules rather than importing the entire package. This approach can help minimize unnecessary overhead and streamline the execution of your function.
+
+For example, if you only need the `Sheets` submodule from the `googleapis` package, you can import it like this:
+
+```typescript
+import { Sheets } from 'npm:@googleapis/sheets'
+```
+
+By selectively importing only the required submodules, you can effectively reduce the size of your function's dependencies and optimize its performance.
+Before finalizing your imports, ensure to review both Deno and NPM dependencies, checking for any unnecessary or redundant dependencies that can be removed. Additionally, check for outdated dependencies and update to the latest versions if possible.
```

Comment:
Move this up to the top `Analyze dependencies` section?

Reply:
Done in https://github.com/supabase/supabase/pull/21403/commits/7a70f6f3c7d04ab0f4c26ebdf8e8b86b171f0ca8